test/unit/specs/components/Personalization/createOnResponseHandler.spec.js

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  PAGE_WIDE_SCOPE_DECISIONS,
+  PAGE_WIDE_SCOPE_DECISIONS_WITH_DOM_ACTION_SCHEMA_ITEMS,
+  PAGE_WIDE_SCOPE_DECISIONS_WITHOUT_DOM_ACTION_SCHEMA_ITEMS
+} from "./responsesMock/eventResponses";
+import createOnResponseHandler from "../../../../../src/components/Personalization/createOnResponseHandler";
+
+describe("Personalization::onResponseHandler", () => {
+  const response = {};
+  let extractDecisions;
+  let executeDecisions;
+  let showContainers;
+
+  beforeEach(() => {
+    const renderableDecisions = PAGE_WIDE_SCOPE_DECISIONS_WITH_DOM_ACTION_SCHEMA_ITEMS;
+    const decisions = PAGE_WIDE_SCOPE_DECISIONS_WITHOUT_DOM_ACTION_SCHEMA_ITEMS;
+    const unprocessedDecisions = PAGE_WIDE_SCOPE_DECISIONS;
+    extractDecisions = jasmine
+      .createSpy("extractDecisions")
+      .and.returnValue([renderableDecisions, decisions, unprocessedDecisions]);
+    executeDecisions = jasmine.createSpy("executeDecisions");
+    showContainers = jasmine.createSpy("showContainers");
+  });
+
+  it("should execute DOM ACTION decisions and return rest of decisions", () => {
+    const expectedResult = {
+      decisions: PAGE_WIDE_SCOPE_DECISIONS_WITHOUT_DOM_ACTION_SCHEMA_ITEMS
+    };
+    const renderDecisions = true;
+    const onResponse = createOnResponseHandler({
+      extractDecisions,
+      executeDecisions,
+      showContainers
+    });
+
+    const result = onResponse({ renderDecisions, response });
+
+    expect(extractDecisions).toHaveBeenCalledWith(response);
+    expect(showContainers).toHaveBeenCalled();
+    expect(executeDecisions).toHaveBeenCalled();
```

Comment:
Assert that `executeDecisions` was called with `renderableDecisions`.

Reply:
thank you